### PR TITLE
chore: Implement v2 Timeout rpc handler

### PR DESCRIPTION
**Pull request by bznein**
## Description

closes: #7355

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/04-channel/v2/keeper/msg_server_test.go [hunk 1]
```diff
@@ -360,7 +361,7 @@ func (suite *KeeperTestSuite) TestMsgAcknowledgement() {
 		{
 			name: "success: NoOp",
 			malleate: func() {
-				suite.chainA.App.GetIBCKeeper().ChannelKeeperV2.SetPacketCommitment(suite.chainA.GetContext(), recvPacket.SourceChannel, recvPacket.Sequence, []byte{})
+				suite.chainA.App.GetIBCKeeper().ChannelKeeperV2.DeletePacketCommitment(suite.chainA.GetContext(), recvPacket.SourceChannel, recvPacket.Sequence)
```
Comment (bznein) [inline modules/core/04-channel/v2/keeper/msg_server_test.go:364]: This is unrelated but I caught it while editing the test below. While the previous also works, this is cleaner

File: modules/core/04-channel/v2/keeper/msg_server_test.go [hunk 2]
```diff
@@ -471,3 +472,112 @@ func (suite *KeeperTestSuite) TestMsgAcknowledgement() {
...
+			name: "failure: callback fails",
+			malleate: func() {
+				path.EndpointA.Chain.GetSimApp().MockModuleV2A.IBCApp.OnTimeoutPacket = func(context.Context, string, string, channeltypesv2.PacketData, sdk.AccAddress) error {
+					return errors.New("OnTimeoutPacket callback failed")
+				}
+			},
+			expError: errors.New("OnTimeoutPacket callback failed"),
+		},
+		{
+			name: "failure: counterparty not found",
```
Comment (AdityaSripal) [inline modules/core/04-channel/v2/keeper/msg_server_test.go:522]:
```suggestion
			name: "failure: channel not found",
```

**Review by AdityaSripal — APPROVED** [review]
Nice work!!

File: modules/core/04-channel/v2/keeper/msg_server_test.go [hunk 3]
```diff
@@ -433,3 +435,112 @@ func (suite *KeeperTestSuite) TestMsgAcknowledgement() {
...
+			suite.SetupTest()
+
+			path = ibctesting.NewPath(suite.chainA, suite.chainB)
+			path.SetupV2()
+
+			// Send packet from A to B
+			timeoutTimestamp := suite.chainA.GetTimeoutTimestamp()
+
+			mockData := mockv2.NewMockPacketData(mockv2.ModuleNameA, mockv2.ModuleNameB)
+			msgSendPacket := channeltypesv2.NewMsgSendPacket(path.EndpointA.ChannelID, timeoutTimestamp, suite.chainA.SenderAccount.GetAddress().String(), mockData)
```
Comment (chatton) [inline modules/core/04-channel/v2/keeper/msg_server_test.go:518]: we should be able to use the new helper fn for this now (might need to merge feature branch first)

**Review by chatton — APPROVED** [review]
LGTM just left a nit, we can address the `endpoint.MsgTimeout` and update the test to use `endpoint.MsgSendPacket` in a follow up if you want!

**Comment by bznein** [comment]
We realized with @chatton that something was wrong in the timeout. I'm gonna apply some changes and send for review again

File: modules/core/04-channel/v2/keeper/msg_server_test.go [hunk 4]
```diff
@@ -433,3 +434,97 @@ func (suite *KeeperTestSuite) TestMsgAcknowledgement() {
...
+			},
+			expError: channeltypesv2.ErrInvalidPacket,
+		},
+		{
+			name: "failure: unable to timeout if packet has been received",
+			malleate: func() {
+				err := path.EndpointB.MsgRecvPacket(packet)
+				suite.Require().NoError(err)
+			},
+			expError: commitmenttypes.ErrInvalidProof,
```
Comment (chatton) [inline modules/core/04-channel/v2/keeper/msg_server_test.go:496]: added this test to ensure timing out is not possible if the packet has been received

File: proto/ibc/core/channel/v2/tx.proto [hunk 5]
```diff
@@ -117,11 +117,10 @@ message MsgTimeout {
 
   option (gogoproto.goproto_getters) = false;
 
-  Packet                    packet             = 1 [(gogoproto.nullable) = false];
-  bytes                     proof_unreceived   = 2;
-  ibc.core.client.v1.Height proof_height       = 3 [(gogoproto.nullable) = false];
-  uint64                    next_sequence_recv = 4;
```
Comment (chatton) [inline proto/ibc/core/channel/v2/tx.proto:123]: removed this field as it was only required for Ordered channels which are not a thing now

**Pull request merged**